addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js

```diff
@@ -271,6 +275,103 @@ class EnvironmentScService extends Service {
     return result;
   }
 
+  async changeWorkspaceRunState(requestContext, { id, operation }) {
+    const existingEnvironment = await this.mustFind(requestContext, { id });
+
+    // Make sure the user has permissions to change the environment run state
+    await this.assertAuthorized(
+      requestContext,
+      { action: 'update-sc', conditions: [this._allowAuthorized] },
+      existingEnvironment,
+    );
+
+    const { status, outputs, projectId } = existingEnvironment;
+
+    // expected environment run state based on operation
+    let expectedStatus;
+    switch (operation) {
+      case 'start':
+        expectedStatus = 'STOPPED';
+        break;
+      case 'stop':
+        expectedStatus = 'COMPLETED';
+        break;
+      default:
+        throw this.boom.badRequest(`operation ${operation} is not valid, only "start" and "stop" are supported`, true);
+    }
+
+    if (status !== expectedStatus) {
+      throw this.boom.badRequest(
+        `unable to ${operation} environment with id "${id}" - current status "${status}"`,
+        true,
+      );
+    }
+    let instanceType;
+    let instanceIdentifier;
+    const outputsObject = cfnOutputsArrayToObject(outputs);
+    // TODO: Remove MetaConnection1Type check to include RStudio after CNAME patch
+    if ('Ec2WorkspaceInstanceId' in outputsObject && !('MetaConnection1Type' in outputsObject)) {
```

Comment:
Same comment applies here. Having the 'MetaConnection1Type' output available might not necessarily mean it's an RStudio environment. The value for this needs to be 'RStudio'. 
(Users can add templates with this output for a non-RStudio workspace type.)